docs/databags.md

```diff
@@ -93,7 +93,10 @@ An apps hash is a contextual (see above) Hash, part of a plaintext data bag item
 * `['bag']` - A string that points to an externalized Apps Hash in which all keys (except this one) are valid. 
 * `[app]` - The name of an app to manage (disk name)
 * `[app]['remove']` - If true, remove this app instead of creating / managing  (default - false)
-* `[app]['local']` - If true, manage files in the local directories instead of the "default" (default-false)
+* `[app]['local']` - If true, manage files in the local directories instead of the "default" (default - false, but forced true if download-url is specified)
```

Comment:
Splunk apps have two different folders for configuration. app_folder/default/ and app_folder/local/ The intent being that apps are shipped with "default" configuration, and people installing the apps (and those using the webui) can override / modify configurations though the local directory. (This precedence is described in the [splunk docs](http://docs.splunk.com/Documentation/Splunk/latest/Admin/Wheretofindtheconfigurationfiles) ) 

When this is being used by Chef to generate an app from scratch, we would want to write our configurations to default (as people using the webui would want to modify things freely). When we are modifying configuration for an app from a tarball, or an app that's pre-installed with the core Splunk package, any customizations we manage from Chef we would want to write to the local directory (the default directory being what is shipped with the app itself). 

We can't distinguish if we're working with a pre-installed app or an app generated by Chef (since on subsequent chef-client runs they would look identical), however an app that we install via Tarball, we would know that we must use the local directory.